src/features/listVisualizer/tree/Tree.tsx

```diff
@@ -0,0 +1,459 @@
+import { Layer, Line } from 'react-konva';
+
+import { Config } from '../Config';
+import { Data, Pair } from '../ListVisualizerTypes';
+import { head, isFunction, isPair, tail } from '../ListVisualizerUtils';
+import { DataTreeNode } from './DataTreeNode';
+import { PairTreeNode } from './PairTreeNode';
+import { DrawableTreeNode, FunctionTreeNode, TreeNode } from './TreeNode';
+
+/**
+ *  A tree object built based on a list or pair.
+ */
+export class Tree {
+  private _rootNode: PairTreeNode;
+  private nodes: DrawableTreeNode[];
+
+  /**
+   * Constructs a tree given a root node and a list of nodes.
+   * @param {PairTreeNode} rootNode The root node of the tree.
+   * @param {DrawableTreeNode[]} nodes The memoized nodes of the tree in list form.
```

Comment:
The `{PairTreeNode}` and `{DrawableTreeNode[]}` in the function documentation doesn't really do anything